FarihaIS/VanRoomies
# feat(maps): Add Location Editing, Map View, and Automatic Posted Date

# Welcome to VanRoomies! 👋✈️

Fixes: #125 
Fixes: #145
Fixes: #146
Fixes: #147

## Description of the change:
- Added MapsActivity accessed by a button in ViewListingActivity. Shows the coordinates set for the listing's location field
- Added a button to update listing location
- Added optional fields in listing creation to add location. Default: (0.0000, 0.0000)
- Changed Posted Date to automatically be set to the current date when creating a new listing

## How to manually test:
1. Run MongoDB and the backend server
2. Start the app
3. Create a listing with no lat/long. They should default to (0.0000, 0.0000)
4. Create a listing with lat and/or long out of bounds. The app checks that -180 < x < 180 where x can be lat or long
5. Create a listing with Latitude: `45.425507` and Longitude: `-75.700233`. The listing should round to 4 decimal places and the map will show the Ottawa Parliament building.
6. Go into the created listings and try editing the location coordinates. The map should update accordingly as well
